src/renderer/util/index.ts

```diff
@@ -33,6 +34,26 @@ const canUserRead = async (filePath: string): Promise<boolean> => {
   }
 };
 
+async function determineFilesFromNestedPath(
+  path: string
+): Promise<string[]> {
+  const isMultifile = determineIsMultifile(path);
+  const stats = await fsPromises.stat(path);
+  if (!stats.isDirectory() || isMultifile) {
+    return [path];
+  }
+  const canRead = await canUserRead(path);
+  if (!canRead) {
+    throw new Error(`User does not have permission to read ${path}`);
+  }
+  const pathsUnderFolder = await fsPromises.readdir(path, {
+    withFileTypes: true,
+  });
+  return pathsUnderFolder
```

Comment:
So if it is not a regular file or a multifile, return a list of every path in the dir, ignoring subdirs?

Reply:
To be 100% clear - I didn't write this exact function, I just extracted it  from the below function of similar name because I thought it would look cleaner, though that was ultimately irrelevant to what I ended up doing.

But to answer your question, yeah, that's the existing behavior. If it's not a regular file or a multifile, then it it's a directory, and (for some reason) we only grab the top-level files in a given directory and not any in its sub-dirs. IDK if we want that long-term, but it's what we have.